sw/device/lib/crypto/drivers/aes.c

```diff
@@ -67,56 +76,96 @@ aes_error_t aes_begin(aes_params_t params) {
       return kAesInternalError;
   }
 
+  uint32_t share0 = kBase + AES_KEY_SHARE0_0_REG_OFFSET;
+  uint32_t share1 = kBase + AES_KEY_SHARE1_0_REG_OFFSET;
+
+  for (size_t i = 0; i < key_words; ++i) {
+    abs_mmio_write32(share0 + i * sizeof(uint32_t), key.key_shares[0][i]);
+    abs_mmio_write32(share1 + i * sizeof(uint32_t), key.key_shares[1][i]);
+  }
+  for (size_t i = key_words; i < 8; ++i) {
+    // NOTE: all eight share registers must be written; in the case we don't
+    // have enough key data, we fill it with zeroes.
+    abs_mmio_write32(share0 + i * sizeof(uint32_t), 0);
+    abs_mmio_write32(share1 + i * sizeof(uint32_t), 0);
+  }
+  return spin_until(AES_STATUS_IDLE_BIT);
+}
+
+/**
+ * Configure the AES block and write the key and IV if applicable.
+ *
+ * @param key AES key.
+ * @param iv IV to use (ignored if the mode does not require an IV).
+ * @param encrypt True for encryption, false for decryption.
+ * @return result, OK or error.
+ */
+static aes_error_t aes_begin(aes_key_t key, const uint32_t iv[4],
+                             bool encrypt) {
```

Comment:
Use `hardened_bool_t` instead. On the hardware side this is protected using shadow registers, but we should still consider hardening the configuration parameter in software.